### PR TITLE
update TTL doc to allow split tasks for utf8 column

### PR DESCRIPTION
**Pull request by lcwangchao**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/18713
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by YangKeao — APPROVED** [review]

File: time-to-live.md [hunk 1]
```diff
@@ -243,7 +243,11 @@ TTL 功能能够与 TiDB 的迁移、备份、恢复工具一同使用。
 * 不允许在临时表上设置 TTL 属性，包括本地临时表和全局临时表。
 * 具有 TTL 属性的表不支持作为外键约束的主表被其他表引用。
 * 不保证所有过期数据立即被删除，过期数据被删除的时间取决于后台清理任务的调度周期和调度窗口。
-* 对于使用[聚簇索引](/clustered-indexes.md)的表，如果主键的类型不是整数类型或二进制字符串类型，TTL 任务将无法被拆分成多个子任务。这将导致 TTL 任务只能在一个 TiDB 节点上按顺序执行。如果表中的数据量较大，TTL 任务的执行可能会变得缓慢。
+* 对于使用[聚簇索引](/clustered-indexes.md)的表，仅以下场景支持将 TTL 任务拆分成多个子任务：
+    - 主键或者复合主键的第一列为整数或二进制字符串类型。
```
Comment (hfxsd) [inline time-to-live.md:247]:
```suggestion
    - 主键或者复合主键的第一列为整数或二进制字符串类型。其中，二进制字符串类型主要指下面几种：
        - `CHAR(N) CHARACTER SET BINARY`
        - `VARCHAR(N) CHARACTER SET BINARY`
        - `BINARY(N)`
        - `VARBINARY(N)`
```

File: time-to-live.md [hunk 2]
```diff
@@ -243,7 +243,11 @@ TTL 功能能够与 TiDB 的迁移、备份、恢复工具一同使用。
...
+    - 主键或者复合主键的第一列的字符集为 `utf8` 或者 `utf8mb4` 且 COLLATE 设置为 `utf8_bin`、 `utf8mb4_bin` 或者 `utf8mb4_0900_bin`。
+* 对于不支持拆分 TTL 子任务的表，TTL 任务只能在一个 TiDB 节点上按顺序执行。此时如果表中的数据量较大，TTL 任务的执行可能会变得缓慢。
```
Comment (hfxsd) [inline time-to-live.md:249]:
```suggestion
* 对于不支持拆分 TTL 子任务的表，意味着扫描阶段使用的是 [TableFullScan 算子](/explain-overview.md#算子简介)（如果能够拆分子任务，意味着扫描阶段使用的是 TableRangeScan 算子），TTL 任务只能在一个 TiDB 节点上按顺序执行。此时如果表中的数据量较大，TTL 任务的执行可能会变得缓慢。
```

Comment (dulao5) [reply time-to-live.md:249]: @hfxsd 

> 意味着扫描阶段使用的是 [TableFullScan 算子](/explain-overview.md#算子简介)（如果能够拆分子任务，意味着扫描阶段使用的是 TableRangeScan 算子），

可以在前面加上 「如果 TTL 的条件字段没有索引的话，」 这样更严谨一些。

Comment (lcwangchao) [reply time-to-live.md:249]: 其实对于不支持拆分的单个 TTL 子任务，也不是说一定会使用 TableFullScan，这个还是取决于优化器的优化。比如用户创建了某些索引，当优化器判断当前查询用索引更快的时候，也会使用索引。

所以能否不强调用的是什么算子，只是阐述单个任务扫描和多个任务并发扫描的区别

File: time-to-live.md (hunk 1 above)
Comment (lcwangchao) [reply time-to-live.md:247]: 也可以加上 `BIT(N)`

File: time-to-live.md [hunk 3]
```diff
@@ -243,7 +243,16 @@ TTL 功能能够与 TiDB 的迁移、备份、恢复工具一同使用。
 * 不允许在临时表上设置 TTL 属性，包括本地临时表和全局临时表。
 * 具有 TTL 属性的表不支持作为外键约束的主表被其他表引用。
 * 不保证所有过期数据立即被删除，过期数据被删除的时间取决于后台清理任务的调度周期和调度窗口。
-* 对于使用[聚簇索引](/clustered-indexes.md)的表，如果主键的类型不是整数类型或二进制字符串类型，TTL 任务将无法被拆分成多个子任务。这将导致 TTL 任务只能在一个 TiDB 节点上按顺序执行。如果表中的数据量较大，TTL 任务的执行可能会变得缓慢。
+* 对于使用[聚簇索引](/clustered-indexes.md)的表，仅以下场景支持将 TTL 任务拆分成多个子任务：
```
Comment (hfxsd) [inline time-to-live.md:246]:
```suggestion
* 对于使用[聚簇索引](/clustered-indexes.md)的表，仅支持在以下场景中将 TTL 任务拆分成多个子任务：
```

File: time-to-live.md [hunk 4]
```diff
@@ -243,7 +243,16 @@ TTL 功能能够与 TiDB 的迁移、备份、恢复工具一同使用。
...
+    - 主键或者复合主键的第一列为整数或二进制字符串类型。其中，二进制字符串类型主要指下面几种：
+        - `CHAR(N) CHARACTER SET BINARY`
+        - `VARCHAR(N) CHARACTER SET BINARY`
+        - `BINARY(N)`
+        - `VARBINARY(N)`
+        - `BIT(N)`
+    - 主键或者复合主键的第一列的字符集为 `utf8` 或者 `utf8mb4` 且 COLLATE 设置为 `utf8_bin`、 `utf8mb4_bin` 或者 `utf8mb4_0900_bin`。
+* 对于不支持拆分 TTL 子任务的表，TTL 任务只能在一个 TiDB 节点上按顺序执行。此时如果表中的数据量较大，TTL 任务的执行可能会变得缓慢。
+* 对于主键第一列类型是 `utf8` 或者 `utf8mb4` 的表，仅会根据 ASCII 可见字符的范围进行子任务拆分。如果大量的主键值具有相同的 ASCII 前缀，可能会造成任务拆分的不均匀。
```
Comment (hfxsd) [inline time-to-live.md:255]:
```suggestion
```

File: time-to-live.md [hunk 5]
```diff
@@ -243,7 +243,16 @@ TTL 功能能够与 TiDB 的迁移、备份、恢复工具一同使用。
...
-* 对于使用[聚簇索引](/clustered-indexes.md)的表，如果主键的类型不是整数类型或二进制字符串类型，TTL 任务将无法被拆分成多个子任务。这将导致 TTL 任务只能在一个 TiDB 节点上按顺序执行。如果表中的数据量较大，TTL 任务的执行可能会变得缓慢。
+* 对于使用[聚簇索引](/clustered-indexes.md)的表，仅以下场景支持将 TTL 任务拆分成多个子任务：
+    - 主键或者复合主键的第一列为整数或二进制字符串类型。其中，二进制字符串类型主要指下面几种：
+        - `CHAR(N) CHARACTER SET BINARY`
+        - `VARCHAR(N) CHARACTER SET BINARY`
+        - `BINARY(N)`
+        - `VARBINARY(N)`
+        - `BIT(N)`
+    - 主键或者复合主键的第一列的字符集为 `utf8` 或者 `utf8mb4` 且 COLLATE 设置为 `utf8_bin`、 `utf8mb4_bin` 或者 `utf8mb4_0900_bin`。
+* 对于不支持拆分 TTL 子任务的表，TTL 任务只能在一个 TiDB 节点上按顺序执行。此时如果表中的数据量较大，TTL 任务的执行可能会变得缓慢。
```
Comment (hfxsd) [inline time-to-live.md:254]:
```suggestion
* 对于主键第一列类型是 `utf8` 或者 `utf8mb4` 的表，仅会根据 ASCII 可见字符的范围进行子任务拆分。如果大量的主键值具有相同的 ASCII 前缀，可能会造成任务拆分不均匀。
* 对于不支持拆分 TTL 子任务的表，TTL 任务只能在一个 TiDB 节点上按顺序执行。此时如果表中的数据量较大，TTL 任务的执行可能会变得缓慢。
```

File: time-to-live.md [hunk 6]
```diff
@@ -243,7 +243,16 @@ TTL 功能能够与 TiDB 的迁移、备份、恢复工具一同使用。
...
 * 不保证所有过期数据立即被删除，过期数据被删除的时间取决于后台清理任务的调度周期和调度窗口。
-* 对于使用[聚簇索引](/clustered-indexes.md)的表，如果主键的类型不是整数类型或二进制字符串类型，TTL 任务将无法被拆分成多个子任务。这将导致 TTL 任务只能在一个 TiDB 节点上按顺序执行。如果表中的数据量较大，TTL 任务的执行可能会变得缓慢。
+* 对于使用[聚簇索引](/clustered-indexes.md)的表，仅支持在以下场景中将 TTL 任务拆分成多个子任务：
+    - 主键或者复合主键的第一列为整数或二进制字符串类型。其中，二进制字符串类型主要指下面几种：
+        - `CHAR(N) CHARACTER SET BINARY`
+        - `VARCHAR(N) CHARACTER SET BINARY`
+        - `BINARY(N)`
+        - `VARBINARY(N)`
+        - `BIT(N)`
+    - 主键或者复合主键的第一列的字符集为 `utf8` 或者 `utf8mb4` 且 COLLATE 设置为 `utf8_bin`、 `utf8mb4_bin` 或者 `utf8mb4_0900_bin`。
```
Comment (lilin90) [inline time-to-live.md:253]:
```suggestion
    - 主键或者复合主键的第一列的字符集为 `utf8` 或者 `utf8mb4`，且 COLLATE 设置为 `utf8_bin`、 `utf8mb4_bin` 或者 `utf8mb4_0900_bin`。
```

File: time-to-live.md [hunk 7]
```diff
@@ -243,7 +243,16 @@ TTL 功能能够与 TiDB 的迁移、备份、恢复工具一同使用。
...
+* 对于主键第一列类型是 `utf8` 或者 `utf8mb4` 的表，仅会根据 ASCII 可见字符的范围进行子任务拆分。如果大量的主键值具有相同的 ASCII 前缀，可能会造成任务拆分不均匀。
```
Comment (lilin90) [inline time-to-live.md:254]:
```suggestion
* 对于主键第一列字符集类型是 `utf8` 或者 `utf8mb4` 的表，仅会根据 ASCII 可见字符的范围进行子任务拆分。如果大量的主键值具有相同的 ASCII 前缀，可能会造成任务拆分不均匀。
```

**Review by hfxsd — APPROVED** [review]

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**